1_prep.R

```diff
@@ -17,36 +17,25 @@ p1 <- list(
              readr::read_csv(p1_lake_cell_tile_xwalk_csv, col_types=cols()) %>%
                filter(site_id %in% p1_nml_site_ids) %>%
                arrange(site_id)),
+  # NetCDF files with munged GCM driver data (one per GCM)
+  # files copied from lake-temperature-model-prep repo
+  tar_target(p1_gcm_ncs, {
+    filename <- sprintf('1_prep/in/GCM_%s.nc', p1_gcm_names)
+    return(filename)
+  }, format = 'file', pattern = map(p1_gcm_names)),
 
   # Define mapping variables
   tar_target(p1_site_ids, p1_lake_cell_tile_xwalk_df %>% pull(site_id)),
   tar_target(p1_cell_nos, unique(p1_lake_cell_tile_xwalk_df %>% pull(data_cell_no))),
   tar_target(p1_gcm_names, c('ACCESS', 'GFDL', 'CNRM', 'IPSL', 'MRI', 'MIROC5')),
   tar_target(p1_gcm_dates, c('1980_1999', '2040_2059', '2080_2099')),
   
-  # COMMENTING OUT FOR NOW, WHILE WE REFINE NETCDF APPROACH
-  # # map over gcm names to read in netCDF files - one per GCM
-  # tar_target(p1_gcm_ncs, {
-  #   filename <- sprintf('1_prep/in/7_GCM_%s.nc', p1_gcm_names)
-  #   return(filename)
-  # }, format = 'file', pattern = map(p1_gcm_names)),
-  # 
-  
-  # # TODO - come up with a more efficient way to split netCDF files
-  # # right now, each netCDF file is read in many times
-  # # split netCDF into feather files - by cell, GCM, and time period
-  # tar_target(p1_meteo_feathers,
-  #            munge_nc_files(p1_gcm_ncs, p1_gcm_names, p1_cell_nos, p1_gcm_dates, 
-  #                           outfile_template = '1_prep/out/GCM_%s_%s_%s.feather'),
-  #            format = 'file',
-  #            pattern = cross(p1_cell_nos, map(p1_gcm_ncs, p1_gcm_names), p1_gcm_dates)),
-  
-  # FOR NOW, USE LINDSAY'S FEATHER FILES, BROUGHT IN MANUALLY
-  # mapping over gcm_names, gcm_dates, and cell_nos to read in Lindsay's created feather files
-  tar_target(p1_meteo_feathers, 
-             sprintf('1_prep/tmp/GCM_%s_%s_%s.feather', p1_gcm_names, p1_gcm_dates, p1_cell_nos),
+  # Generate GCM/time-period/cell-specific feather files
```

Comment:
This is a lot of file i/o. What if we did this munging per tile instead? That would significantly cut down on the number of files and hopefully be a nice intermediate between munging everything in the nc file vs munging each cell.

Reply:
Hmm I was leaving it as we had set it up, since our initial approach depended on having a single meteo file per model run. But I agree it's a lot a lot of files and definitely not great for scaling up. Changing to a per tile file format would mean changing how we set up our [model config](https://github.com/USGS-R/lake-temperature-process-models/blob/main/1_prep/src/build_model_config.R) and then reading in a larger meteo file [here](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L12) and then filtering to the correct cell. Then the data for a given gcm for a given cell for a given time period would only ever [be saved](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L110) to the simulation directory, meaning it would be deleted if the model runs successfully. Seems like that approach could work well, so I'll add it to this PR and test it out.

Reply:
Ahhh right. So this is potentially a lot more work that I had originally thought it would be to make that change. Though, might be worth it so that we can handle the midwest scale. Although, should we have any reason to be concerned about multiple lakes trying to access/read a single tile feather at the same time when we parallelize by lakes? I guess a similar thing was happening for multiple cells within one lake, but there aren't as many lakes sharing one cell as there would be lakes sharing one tile file.

Reply:
It's not too much more work - I have it coded up now. But your point about multiple lakes trying to access a single tile feather when we parallelize is a good one -- I have no sense of whether or not that would be an issue, though we could run some tests on Denali. Another factor, I realized, is that in order to parse the meteo data to the correct cell [here](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L12), we'll need the model_config to include a column for `data_cell_no`, so that we could pass it to `add_burn_in_out_to_meteo()`. Previously we'd [excluded `data_cell_no` from the model config](https://github.com/USGS-R/lake-temperature-process-models/blob/main/1_prep/src/build_model_config.R#L43) because we didn't want any change in the numbering approach for cells to trigger rebuilds. Though I'm realizing that with either proposed approach, either `data_cell_no` or `data_tile_no` would be captured in the `raw_meteo_fl` field that is passed to model execution, so if that filename changed it would also trigger rebuilds, right?

Reply:
Yes, if the filename changed then we would end up triggering rebuilds

Reply:
Ok - Lindsay and I chatted about this today and came up with the following plan:
  1) Revert back to the gcm/time-period/cell-specific approach (generating a single file for each gcm, for each time-period, for each cell) until we are more clear on the implications of many model runs needing  to access the same file when we are running in parallel, which would happen with the gcm/time-period/tile approach
  2) Modify our handling of the meteo files, such that we add burn-in and burn-out immediately upon pulling the data from the netCDF, to reduce file I/O
  3) Write the meteo data to .csv after pulling it from the netCDFs, so that we can copy the files into the simulation directories, rather than reading in the data from feather files and then writing to csv within the simulation directories

Reply:
Done!